### PR TITLE
feat: add C implementation for `math/base/special/acsc`

### LINKED ISSUE #859: [RFC]: Add C implementation for `@stdlib/math/base/special/acsc`
### Description

This RFC proposes adding native C implementation for [`@stdlib/math/base/special/acsc`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/acsc/).

```c
double stdlib_base_acsc( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves #859  .

## Description

> What is the purpose of this pull request?

This pull request:

-    add C implementation for math/base/special/acsc

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #859 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

**Comment by Userf1995** [comment]
@Pranavchiku @Planeshifter @kgryte 
"Thank you for bringing this to my attention! I understand that there's already an existing pull request addressing a similar implementation, but it seems that it doesn't entirely resolve the issue, especially considering this package's purpose of facilitating the implementation of other C packages. With that in mind, I've gone ahead and implemented the necessary changes. I would greatly appreciate your feedback and review on the adjustments made."

File: lib/node_modules/@stdlib/math/base/special/acsc/src/main.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+*/
+
+#include "stdlib/math/base/special/acsc.h"
+#include "stdlib/math/base/special/asin.h"
+
+/**
+* Computes the arccosecant of a  number.
+*
+* @param x    input value
+* @return	  output value
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsc/src/main.c]:
```suggestion
* @param x    input value
* @return     output value
```

File: lib/node_modules/@stdlib/math/base/special/acsc/src/main.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+*
+* @example
+* double out = stdlib_base_acsc( 1.0 );
+* // returns ~1.57
+*/
+double stdlib_base_acsc( const double x ) {
+    return stdlib_base_asin( 1.0 / x );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsc/src/main.c]:
```suggestion
	return stdlib_base_asin( 1.0 / x );
```

File: lib/node_modules/@stdlib/math/base/special/acsc/test/test.native.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsc/test/test.native.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @Rejoan-Sardar !

**Pull request merged**